zebra-state/src/service/read/find.rs

```diff
@@ -53,7 +91,71 @@ where
         || db.contains_hash(hash)
 }
 
-/// Find the first hash that's in the peer's `known_blocks` and the chain.
+/// Create a block locator from `chain` and `db`.
+///
+/// A block locator is used to efficiently find an intersection of two node's chains.
+/// It contains a list of block hashes at decreasing heights, skipping some blocks,
+/// so that any intersection can be located, no matter how long or different the chains are.
+pub fn block_locator<C>(chain: Option<C>, db: &ZebraDb) -> Option<Vec<block::Hash>>
+where
+    C: AsRef<Chain>,
+{
+    let chain = chain.as_ref();
+
+    let tip_height = tip_height(chain, db)?;
+
+    let heights = block_locator_heights(tip_height);
+    let mut hashes = Vec::with_capacity(heights.len());
+
+    for height in heights {
+        if let Some(hash) = hash_by_height(chain, db, height) {
+            hashes.push(hash);
+        }
+    }
+
+    Some(hashes)
+}
+
+/// Get the heights of the blocks for constructing a block_locator list.
+///
+/// Zebra uses an exponentially decreasing list of block heights, starting at the tip.
+/// See [`block_locator()`] for details.
+pub fn block_locator_heights(tip_height: block::Height) -> Vec<block::Height> {
+    // The initial height in the returned `vec` is the tip height,
+    // and the final height is `MAX_BLOCK_REORG_HEIGHT` below the tip.
+    //
+    // The initial distance between heights is 1. This distance increases
+    // exponentially with base 2 between each subsequent height.
```

Comment:
I even found a simpler way to put it.
```suggestion
    // The initial distance between heights is 1, and it doubles
    //  between each subsequent height.
```

Reply:
I fixed this in 9efa75640.

Since this change isn't blocking, and isn't related to this PR, let's open another PR for any further changes?

Reply:
Definitely.